File: wrappers/csharp/Intel.RealSense/Frames/DepthFrame.cs
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2017 Intel Corporation. All Rights Reserved.

namespace Intel.RealSense
{
    using System;
    using System.Collections.Generic;
    using System.Runtime.InteropServices;

    public class DepthFrame : VideoFrame
    {
        public DepthFrame(IntPtr ptr)
            : base(ptr)
        {
        }

        /// <summary>Given the 2D depth coordinate (x,y) provide the corresponding depth in metric units</summary>
        /// <returns>depth in metric units</returns>
        public float GetDistance(int x, int y)
        {
            object error;
            return NativeMethods.rs2_depth_frame_get_distance(Handle, x, y, out error);
        }
    }
}

File: wrappers/android/librealsense/src/main/java/com/intel/realsense/librealsense/DepthFrame.java
package com.intel.realsense.librealsense;

public class DepthFrame extends VideoFrame {
    protected DepthFrame(long handle) {
        super(handle);
        mOwner = false;
    }

    public float getDistance(int x, int y) {
        return nGetDistance(mHandle, x, y);
    }

    public float getUnits() {
        return nGetUnits(mHandle);
    }

    private static native float nGetDistance(long handle, int x, int y);
    private static native float nGetUnits(long handle);
}
